controller/src/drone_state.rs

```diff
@@ -0,0 +1,297 @@
+                        timestamp,
+                    },
+                }),
+            },
+            WorldStateMessage {
+                cluster: msg.cluster.clone(),
+                message: ClusterStateMessage::DroneMessage(DroneMessage {
+                    drone: msg.drone_id.clone(),
+                    message: DroneMessageType::KeepAlive { timestamp },
+                }),
+            },
+        ],
+        DroneStateUpdate::BackendStateMessage(msg) => vec![
+            WorldStateMessage {
+                cluster: msg.cluster.clone(),
+                message: ClusterStateMessage::BackendMessage(BackendMessage {
+                    backend: msg.backend.clone(),
+                    message: BackendMessageType::State {
+                        state: msg.state,
+                        timestamp,
+                    },
+                }),
+            },
+            WorldStateMessage {
+                cluster: msg.cluster.clone(),
+                message: ClusterStateMessage::BackendMessage(BackendMessage {
+                    backend: msg.backend.clone(),
+                    message: BackendMessageType::Assignment {
+                        drone: msg.drone.clone(),
+                    },
+                }),
+            },
+        ],
+    }
+}
+
+pub async fn apply_state_message(
+    nats: &TypedNats,
+    message: &WorldStateMessage,
+) -> Result<Option<u64>> {
+    let overwrite = message.overwrite();
+    tracing::info!(?message, ?overwrite, "Publishing state message.");
+    if overwrite {
+        nats.publish_jetstream(message).await.map(Some)
+    } else {
+        nats.publish_jetstream_if_subject_empty(message).await
+    }
+}
+
+pub async fn monitor_drone_state(nats: TypedNats) -> NeverResult {
+    let mut acme_sub = nats
+        .subscribe(DroneStateUpdate::subscribe_subject_acme())
+        .await?;
+    tracing::info!("Subscribed to ACME DNS messages.");
+
+    let mut connect_sub = nats
+        .subscribe(DroneStateUpdate::subscribe_subject_connect())
+        .await?;
+    tracing::info!("Subscribed to drone connect messages.");
+
+    let mut drone_status_sub = nats
+        .subscribe(DroneStateUpdate::subscribe_subject_drone_status())
+        .await?;
+    tracing::info!("Subscribed to drone status messages.");
+
+    let mut backend_status_sub = nats
+        .subscribe(DroneStateUpdate::subscribe_subject_backend_status())
+        .await?;
+    tracing::info!("Subscribed to backend status messages.");
+
+    loop {
+        let message = select! {
+            acme_msg = acme_sub.next() => acme_msg,
+            connect_msg = connect_sub.next() => connect_msg,
+            drone_status_msg = drone_status_sub.next() => drone_status_msg,
+            backend_status_msg = backend_status_sub.next() => backend_status_msg,
+        };
+
+        if let Some(message) = message {
+            tracing::info!(message=?message.value, "Got state message from drone.");
```

Comment:
This will end up printing out a lot of drone status messages (we had them set to debug level before). Maybe this is fine though? Or we could rework this to have drone status messages be debug level while the others are info level? I'm fine with leaving as is, too